contracts/protocol/v1/proxies/P1LiquidatorProxy.sol

```diff
@@ -0,0 +1,282 @@
+/*
+
+    Copyright 2020 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+import { Ownable } from "@openzeppelin/contracts/ownership/Ownable.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import { BaseMath } from "../../lib/BaseMath.sol";
+import { SignedMath } from "../../lib/SignedMath.sol";
+import { I_PerpetualV1 } from "../intf/I_PerpetualV1.sol";
+import { P1BalanceMath } from "../lib/P1BalanceMath.sol";
+import { P1Types } from "../lib/P1Types.sol";
+
+
+/**
+ * @title P1LiquidatorProxy
+ * @author dYdX
+ *
+ * @notice Proxy contract allowing users to liquidate accounts and pay some extra fees to the
+ * insurance fund.
+ */
+contract P1LiquidatorProxy is
+    Ownable
+{
+    using BaseMath for uint256;
+    using SignedMath for SignedMath.Int;
+    using P1BalanceMath for P1Types.Balance;
+
+    // ============ Events ============
+
+    event LogLiquidatorProxyUsed(
+        address indexed liquidator,
+        address indexed liquidatee,
+        bool isBuy,
+        uint256 liquidationAmount,
+        uint256 feeAmount
+    );
+
+    event LogInsuranceFundSet(
+        address insuranceFund
+    );
+
+    // ============ Immutable Storage ============
+
+    // address of the perpetual contract
+    address public _PERPETUAL_V1_;
+
+    // address of the liquidator contract
+    address public _LIQUIDATOR_;
+
+    // address of the insurance fund
+    address public _INSURANCE_FUND_;
+
+    // base number representing the percentage of fees that go to the insurance fund
```

Comment:
I tried to rephrase/clarify everywhere we used to say “base number”:

`the percentage of fees that go to the insurance fund, fixed-point with 18 decimals`